pkg/usersettings/handlers.go

```diff
@@ -0,0 +1,263 @@
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/proxy"
+	"github.com/openshift/console/pkg/serverutils"
+)
+
+const USER_SETTINGS_NAMESPACE = "openshift-console-user-settings"
+
+var USER_RESOURCE = schema.GroupVersionResource{
+	Group:    "user.openshift.io",
+	Version:  "v1",
+	Resource: "users",
+}
+
+type UserSettingsHandler struct {
+	K8sProxyConfig      *proxy.Config
+	Client              *http.Client
+	Endpoint            string
+	ServiceAccountToken string
+}
+
+func (h *UserSettingsHandler) HandleUserSettings(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+		configMap, err := h.getUserSettings(user)
+		if err != nil {
+			serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to get user settings: %v", err)})
+		} else {
+			serverutils.SendResponse(w, http.StatusOK, configMap)
+		}
+	case http.MethodPost:
+		configMap, err := h.createUserSettings(user)
+		if err != nil {
+			serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to create user settings: %v", err)})
+		} else {
+			serverutils.SendResponse(w, http.StatusOK, configMap)
+		}
+	case http.MethodDelete:
+		err := h.deleteUserSettings(user)
+		if err != nil {
+			serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to delete user settings: %v", err)})
+		} else {
+			w.WriteHeader(http.StatusNoContent)
+		}
+	default:
+		w.Header().Set("Allow", "GET, POST, DELETE")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Unsupported method, supported methods are GET POST DELETE"})
+	}
+}
+
+// Fetch the current user with auth.User token and returning the user-setting ConfigMap.
+func (h *UserSettingsHandler) getUserSettings(user *auth.User) (*core.ConfigMap, error) {
+	context := context.TODO()
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := h.createServiceAccountClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return client.CoreV1().ConfigMaps(USER_SETTINGS_NAMESPACE).Get(context, resourceData.getConfigMapName(), meta.GetOptions{})
+}
+
+// Fetch the current user with auth.User token and create a new user-setting ConfigMap.
+// Returns the existing ConfigMap if it is already exist.
+func (h *UserSettingsHandler) createUserSettings(user *auth.User) (*core.ConfigMap, error) {
+	context := context.TODO()
+
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return nil, err
+	}
+
+	role := &rbac.Role{
```

Comment:
Lets create a helper for this, so we can add unit-test.

Reply:
Done, see helpers.go and helpers_test.go